src/wui/interactive_gamebase.cc

```diff
@@ -299,16 +311,54 @@ void InteractiveGameBase::gamespeed_menu_selected(GameSpeedEntry entry) {
 	}
 }
 
-void InteractiveGameBase::increase_gamespeed() {
+void InteractiveGameBase::increase_gamespeed(uint32_t speed) {
 	if (GameController* const ctrl = get_game()->game_controller()) {
-		ctrl->set_desired_speed(ctrl->desired_speed() + 1000);
+		uint32_t const current_speed = ctrl->desired_speed();
+		switch (speed) {
+		case 250:
+			ctrl->set_desired_speed(current_speed + speed);
+			break;
+		case 1000:
+			if (current_speed < 1000) {
+				ctrl->set_desired_speed(1000);
+			} else {
+				ctrl->set_desired_speed(current_speed - current_speed % speed + speed);
+			}
+			break;
+		case 10000:
+			ctrl->set_desired_speed(
+			   current_speed >= speed ? current_speed - current_speed % speed + speed : speed);
+			break;
+		default:
+			break;
+		}
 	}
 }
 
-void InteractiveGameBase::decrease_gamespeed() {
+void InteractiveGameBase::decrease_gamespeed(uint32_t speed) {
 	if (GameController* const ctrl = get_game()->game_controller()) {
-		uint32_t const speed = ctrl->desired_speed();
-		ctrl->set_desired_speed(1000 < speed ? speed - 1000 : 0);
+		uint32_t const current_speed = ctrl->desired_speed();
+		switch (speed) {
+		case 250:
+			ctrl->set_desired_speed(current_speed > speed ? current_speed - speed : 0);
+			break;
+		case 1000:
+			if (current_speed >= 2000) {
+				ctrl->set_desired_speed(current_speed - current_speed % speed - speed);
+			} else if (current_speed > 1000) {
+				ctrl->set_desired_speed(1000);
+			} else {
+				ctrl->set_desired_speed(0);
+			}
+			break;
+		case 10000:
+			ctrl->set_desired_speed(
+			   current_speed > speed ? current_speed - current_speed % speed - speed : 0);
+			break;
+		default:
+			ctrl->set_desired_speed(1000);
```

Comment:
Setting the speed to 1× does not belong in a function named `decrease`, please move the resetting call to a new function and make this `default` clause here behave like I requested for the `default` in `increase_gamespeed()`

Reply:
done